### PR TITLE
Add peak indicators

### PR DESCRIPTION
**Pull request by michaelgregorius**
Add peak indicators to the mixer strips. They show the maximum peak value that was observed and can be reset by clicking on them.

PeakIndicators

## Implementation details

The implementation works via a signal/slot mechanism. The `Fader` class has a new signal `peakChanged` which reports peak values as amplifications. A new class `PeakIndicator` is added which has a slot `updatePeak` which is connected to the new signal in `Fader`.

The `PeakIndicator` inherits from `QLabel` and mainly deals with updating the label text from the current peak value.

Add a `PeakIndicator` instance to `MixerChannelView`. Add a `reset` method to `MixerChannelView` so that the mixer channel can be reset on the loading of new projects, etc. The current implementation resets the peak indicator back to -inf dbFS. The `reset` method is called in `MixerView::clear`.

Remove the clamping in `Fader::setPeak` so that all peaks are reported. Emit the new signal if the peak changes.

### CODE REVIEW

File: include/MixerChannelView.h [hunk 1]
```diff
@@ -119,6 +123,7 @@ namespace lmms::gui
         QLabel* m_receiveArrow;
         PixmapButton* m_muteButton;
         PixmapButton* m_soloButton;
+        PeakIndicator* m_peakIndicator = 0;
```
Comment (DomClark) [inline include/MixerChannelView.h:126]: We prefer `nullptr` over `0` for a null pointer constant.
```suggestion
        PeakIndicator* m_peakIndicator = nullptr;
```

File: src/gui/widgets/Fader.cpp [hunk 2]
```diff
@@ -207,22 +207,22 @@ void Fader::wheelEvent (QWheelEvent* ev)
...
 			lastPeakTimer.restart();
+			peakChanged(persistentPeak);
 		}
 		update();
 	}
 
 	if (persistentPeak > 0 && lastPeakTimer.elapsed() > 1500)
 	{
 		persistentPeak = qMax<float>(0, persistentPeak-0.05);
+		peakChanged(persistentPeak);
```
Comment (DomClark) [inline src/gui/widgets/Fader.cpp:225]: We prefer to use the `emit` macro when emitting a signal.
```suggestion
			emit peakChanged(persistentPeak);
		}
		update();
	}

	if (persistentPeak > 0 && lastPeakTimer.elapsed() > 1500)
	{
		persistentPeak = qMax<float>(0, persistentPeak-0.05);
		emit peakChanged(persistentPeak);
```

**Review by DomClark — APPROVED** [review]
Generally looks good.

File: include/MixerChannelView.h (hunk 1 above)
Comment (michaelgregorius) [reply include/MixerChannelView.h]: Fixed with commit aea000bc594.

File: src/gui/widgets/Fader.cpp (hunk 2 above)
Comment (michaelgregorius) [reply src/gui/widgets/Fader.cpp]: Fixed with commit aea000bc594.

**Review by zonkmachine — APPROVED** [review]
Tests fine. LGTM!

One thought. I'm guessing that there will be some complaints regarding the size of the indicator numbers and now that the mixer can expand, maybe it could be good (in a later PR...) to let it expand a bit sideways too and give room for larger numbers/text.

**Comment by michaelgregorius** [comment]
> Tests fine. LGTM!
> 
> One thought. I'm guessing that there will be some complaints regarding the size of the indicator numbers and now that the mixer can expand, maybe it could be good (in a later PR...) to let it expand a bit sideways too and give room for larger numbers/text.

Yes, for larger fonts the mixer strips would have to be able to expand to the side as well. This is how small values currently look:

PeakIndicatorSmallValue

If the size of the font is increased the labels will clip if they do not expand sideways.

**Pull request merged**